File: ruby/grade-school/grade_school.rb
class School
  attr_reader :klasses

  def initialize
    @klasses = Hash.new{ |h, k| h[k] = Array.new }
  end

  def add(name, klass)
    @klasses[klass] = @klasses[klass].push(name).sort # keep the names sorted
  end

  def to_hash
    Hash[@klasses.sort] # sort the klass numbers
  end

  def grade(klass)
    @klasses[klass].sort
  end
end

File: ecmascript/grade-school/grade-school.js
export default class School {
  constructor() {
    this.students = {};
  }

  roster() {
    let clone = {};
    for(let grade in this.students) {
      clone[grade] = [...this.students[grade]];
    }
    return clone;
  }

  add(name, grade) {
    this.students[grade] = this.students[grade] || [];
    this.students[grade].push(name);

    // Keep the list sorted
    this.students[grade] = this.students[grade].sort();
  }

  grade(gradeNumber) {
    return [...(this.students[gradeNumber] || [])];
  }
}
